src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -113,104 +114,85 @@ def compilation_hash(otf_closure: stages.ProgramCall) -> int:
+        )
+        cached = factory.Trait(
+            executor=factory.LazyAttribute(
+                lambda o: otf_compile_executor.CachedOTFCompileExecutor(
+                    otf_workflow=workflow.CachedStep(o.otf_workflow, hash_function=o.hash_function),
+                    name=o.name,
+                )
+            ),
+            name_cached="_cached",
+        )
+        device_type = core_defs.DeviceType.CPU
+        hash_function = compilation_hash
+        otf_workflow = factory.SubFactory(
+            GTFNCompileWorkflowFactory, device_type=factory.SelfAttribute("..device_type")
+        )
+        name = factory.LazyAttribute(
+            lambda o: f"run_gtfn_{o.name_device}{o.name_cached}{o.name_postfix}"
+        )
 
+    executor = factory.LazyAttribute(
+        lambda o: otf_compile_executor.OTFCompileExecutor(otf_workflow=o.otf_workflow, name=o.name)
+    )
+    allocator = next_allocators.StandardCPUFieldBufferAllocator()
 
-gtfn_executor = otf_compile_executor.OTFCompileExecutor(
-    name="run_gtfn", otf_workflow=GTFN_DEFAULT_WORKFLOW
-)
-run_gtfn = otf_compile_executor.OTFBackend(
-    executor=gtfn_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
-)
 
-gtfn_imperative_executor = otf_compile_executor.OTFCompileExecutor(
-    name="run_gtfn_imperative",
-    otf_workflow=gtfn_executor.otf_workflow.replace(
-        translation=gtfn_executor.otf_workflow.translation.replace(use_imperative_backend=True),
-    ),
-)
-run_gtfn_imperative = otf_compile_executor.OTFBackend(
-    executor=gtfn_imperative_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
-)
+run_gtfn = GTFNBackendFactory()
 
-# TODO(ricoh): add API for converting an executor to a cached version of itself and vice versa
-gtfn_cached_executor = otf_compile_executor.CachedOTFCompileExecutor(
-    name="run_gtfn_cached",
-    otf_workflow=workflow.CachedStep(
-        step=gtfn_executor.otf_workflow, hash_function=compilation_hash
-    ),
-)
-run_gtfn_cached = otf_compile_executor.OTFBackend(
-    executor=gtfn_cached_executor,
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
+run_gtfn_imperative = GTFNBackendFactory(
+    name_postfix="_imperative",
+    otf_workflow__translation__use_imperative_backend=True,
 )
 
+run_gtfn_cached = GTFNBackendFactory(cached=True)
 
-run_gtfn_with_temporaries = otf_compile_executor.OTFBackend(
-    executor=otf_compile_executor.OTFCompileExecutor(
-        name="run_gtfn_with_temporaries",
-        otf_workflow=gtfn_executor.otf_workflow.replace(
-            translation=gtfn_executor.otf_workflow.translation.replace(
-                lift_mode=LiftMode.FORCE_TEMPORARIES,
-                temporary_extraction_heuristics=global_tmps.SimpleTemporaryExtractionHeuristics,
-            ),
-        ),
-    ),
-    allocator=next_allocators.StandardCPUFieldBufferAllocator(),
+run_gtfn_with_temporaries = GTFNBackendFactory(
+    name_postfix="_with_temporaries",
```

Comment:
Lets make this a trait.

Reply:
Either the suggestion got jumbled up again or there is a misunderstanding about what factoryboy traits are.

Assuming the former:
You might be talking about switching `lift_mode` and `temporary_extraction_heuristics` based on a trait?

Assuming the latter, you would be talking about something like:
```python
class GTFNBackendFactory(factory.Factory):
    ...
    class Params:
        name_postfix = factory.Trait(
            ... and now what?
        )
        ...
        
# usage
GTFNBackendFactory(name_postfix=True)
```

The effect is to negate the purpose of the `name_postfix` parameter, which is to give custom name postfixes. In either case, this would be an easy post-project cooldown PR. The scope of this PR is now fixed.

Reply:
Not sure why you mention the `name_postfix` parameter. I want something like this:
```suggestion
run_gtfn_with_temporaries = GTFNBackendFactory(use_temporaries=True)
```
because for many purposes the lift_mode and heuristics is a low-levle detail I don't care about as a library user.

Reply:
The github view makes it look like you are talking specifically about the line containing `name_postfix`. As I thought, this turns out to be accidental.

The need to change "lift_mode" and "heuristics" in synch arose only as a result of merging with main after the end of the project. Therefore it remains out of scope for this PR.